### DISCUSSION TITLE
Improve API error handling

### DISCUSSION THREAD

**Discussion by andyjac** [distributeaid/shipment-tracker#535] [Questions]
### 1. Dig into graphql errors types and what they mean/best practices around using them
- What error should we be using when a record isn't found (404 cases)?
- When should the generic `ApolloError` be used?
- In general, what are the best practices for error handling and structure in graphql applications?

### 2. Deduplicate error logic
At the moment we have string literals floating around in various places for error messages. We should:
1. Codify and unify the language we use for error messages (is is `Shipment not found` or `Shipment does not exist`?)
2. Encapsulate these error strings in some sort of structure/class that can be reused

**Comment by coderbyheart** [comment]
See https://github.com/distributeaid/shipment-tracker/blob/84ea3d61a049c243eb26c027c4afdba0bdff8c90/src/resolvers/input-validation/errorsToProblemDetails.ts#L6-L40 for a way to express API errors.

IMO ApolloError schould not be used for internal error, instead EntityNotFound which then is wrapped into problem details.

In the end it depends on how the frontend would like to present errors.

**Comment by coderbyheart** [comment]
This has been added to the REST API: 93a304ae13807cc980ccd4c8c1eb052a51d0692e
